CT-1824 Bugfix for responders not able to upload after approval

stephenrichards opened the pull request:
If a responder has uploaded a document to a flagged case, and that case
has been approved by disclosure, but amends requested by Press or Private,
the responder was not able to upload documents again once the case was
back in drafting due to a condition that prohibited it if there was an approved
approver assignemtn on the case.  This fix removes that prohibition, allowing
responders to upload if the case is in drafting regardless of whether or not the
case has previously been approved.

misaka commented on app/controllers/cases_controller.rb:
```diff
@@ -30,6 +30,7 @@ def index
                 .cases
                 .page(params[:page])
                 .decorate
+    @current_tab_name = 'all_cases'
```
Adding stuff to the controller seems a bit messy to me. Can we use a helper in the view and access the name of the action with `action_name` instead?

misaka replied:
Just checked and `action_name` works from inside the view and returns the name of the action.

stephenrichards committed:
CT-1824 Bugfix for responders not able to upload after approval

If a responder has uploaded a document to a flagged case, and that case
has been approved by disclosure, but amends requested by Press or Private,
the responder was not able to upload documents again once the case was
back in drafting due to a condition that prohibited it if there was an approved
approver assignemtn on the case.  This fix removes that prohibition, allowing
responders to upload if the case is in drafting regardless of whether or not the
case has previously been approved.
